frontend/server/src/SecurityTools.php

```diff
@@ -248,13 +248,65 @@ public static function getGitserverAuthorizationToken(
      * gitserver that is valid for a single problem for 5 minutes.
      *
      * @param array<string, string> $claims
-     * @param string $subject
+     * @param \OmegaUp\DAO\VO\Identities $subject
      * @return string The Bearer authorization token.
      */
     public static function getCourseCloneAuthorizationToken(
         array $claims,
-        string $subject
+        \OmegaUp\DAO\VO\Identities $subject
     ): string {
+        $secretKey = self::getCourseCloneSecretKey();
+        $token = (new \ParagonIE\Paseto\Builder())
+            ->setKey($secretKey)
+            ->setVersion(new \ParagonIE\Paseto\Protocol\Version2())
+            ->setPurpose(\ParagonIE\Paseto\Purpose::local())
+            ->setExpiration(
+                (\DateTime::createFromFormat(
+                    'U',
+                    \OmegaUp\Time::get()
+                ))->add(
+                    new \DateInterval(
+                        'P7D'
+                    )
+                )
+            )
+            ->setIssuer($subject->username)
+            ->setSubject(strval($subject->user_id))
+            ->setClaims($claims);
+        return $token->toString();
+    }
+
+    /**
+     * @return array<string, string>
+     */
+    public static function getDecodedCloneCourseToken(
+        string $token,
+        string $courseAlias
+    ): array {
+        $parser = \ParagonIE\Paseto\Parser()::getLocal(
```

Comment:
```suggestion
        $parser = \ParagonIE\Paseto\Parser::getLocal(
```

oops, esos paréntesis se me fueron.

Reply:
Ah, no probé de esa forma.